Eastern-Research-Group/expert-query
# Feature/95 api gis assessment profile

## Related Issues:
* [EQ-95](https://jira.epa.gov/browse/EQ-95)

## Main Changes:
* Added the ability to query the database from the api. 
  * Includes the ability to download files as csv, tsv, xlsx, and json. Pass in either `?format=xlsx` or `?f=xlsx`
  * Data is streamed in from the database and then streamed out as a file to reduce memory usage
  * Setup to query on all possible values. We can limit this in the future.
* Moved away from sequelize, because the streaming support is not very good. I opted to use [knexjs](https://knexjs.org/) instead.
* Added `VCAP_SERVICES` to the required environment variables.

## Steps To Test:
1. Install new dependencies
2. Start up the app
3. Call the following and verify they work
    * http://localhost:9090/data/assessments?assessmentUnitId=[%22INW0871_T1011A%22,%22INW0871_T1012A%22,%20%22VT08-16L02_02%22]&f=xlsx
    * http://localhost:9090/data/assessments?assessmentUnitId=[%22INW0871_T1011A%22,%22INW0871_T1012A%22,%20%22VT08-16L02_02%22]&f=csv
    * http://localhost:9090/data/assessments?assessmentUnitId=[%22INW0871_T1011A%22,%22INW0871_T1012A%22,%20%22VT08-16L02_02%22]&f=tsv
    * http://localhost:9090/data/assessments?assessmentUnitId=[%22INW0871_T1011A%22,%22INW0871_T1012A%22,%20%22VT08-16L02_02%22]&f=json
    * http://localhost:9090/data/assessments?assessmentUnitId=[%22INW0871_T1011A%22,%22INW0871_T1012A%22,%20%22VT08-16L02_02%22]
    * http://localhost:9090/data/assessments/count?assessmentUnitId=[%22INW0871_T1011A%22,%22INW0871_T1012A%22,%20%22VT08-16L02_02%22]
    * http://localhost:9090/data/assessments/count?id=[1,2,3,4,5,6,7,8,9,10]

## TODO:
- [ ] Will need to do some additional testing on cloud.gov. May need to increase available memory.
- [ ] I want to add some more documentation to this code.
